### PR TITLE
Update 9-design.md

### PR DESCRIPTION
**Pull request by achillet3**

### CODE REVIEW

File: template/9-design.md
Comment (christinehahaa) [inline template/9-design.md]:
```suggestion
The app will be developed in Kotlin, optimized for Android, utilizing the Model-View-ViewModel (MVVM)^architecture. This approach ensures a clear separation of concerns, making the management of the app’s UI and business logic more efficient and maintainable.
```

File: template/9-design.md
Comment (christinehahaa) [inline template/9-design.md:14]: Not sure if we should add Nominatim API for location searching?

File: template/9-design.md
Comment (christinehahaa) [inline template/9-design.md:28]: perhaps mention nav bar?

File: template/9-design.md
Comment (nikhen-s) [inline template/9-design.md]:
```suggestion
- **Bio:** Short description of a user. (String)
```

File: template/9-design.md
Comment (christinehahaa) [inline template/9-design.md]:
```suggestion
- **Components**: Chat list and My Friends list
```

File: template/9-design.md
Comment (nikhen-s) [inline template/9-design.md]: Nit: Maybe we could change all of these from plural to singular: e.g. Names to name, email addresses to email address, etc to ensure consistency.

File: template/9-design.md
Comment (nikhen-s) [inline template/9-design.md]:
```suggestion
- **Organizer List**: List of user ids of users organizing events. (Array of userIds)
```

File: template/9-design.md
Comment (christinehahaa) [inline template/9-design.md]:
```suggestion
- **Description**: The Messaging Screen serves as a central hub for users to engage in real-time conversations. It provides a comprehensive view of all conversations, allowing users to seamlessly switch between chats. Additionally, a tab allows users to specifically view their friends list, streamlining the ability to review the connections made and communications with close contacts.
```

File: template/9-design.md
Comment (nikhen-s) [inline template/9-design.md]: Do we need load balancing for the MVP initially? Maybe this is when we have a larger user base.

File: template/9-design.md
Comment (christinehahaa) [inline template/9-design.md]:
```suggestion
- **Application Logic**: Manages user authentication, registration, profile updates, and adding friends via QR scan.
```

File: template/9-design.md
Comment (nikhen-s) [inline template/9-design.md:188]:
```suggestion
- Tests verify the logic and behavior of ViewModel classes, including interactions with factory-created UI components.
- Tests utilize Mock Repositories for the databases to isolate testing specifically on the ViewModel.
```

File: template/9-design.md
Comment (christinehahaa) [inline template/9-design.md:14]: also need to add something about 3rd party payment api?

**Review by nikhen-s — CHANGES_REQUESTED** [review]
Just small nits, other than that LGTM, well done!

File: template/9-design.md
Comment (christinehahaa) [inline template/9-design.md:80]: add something about payment API?

**Review by christinehahaa — CHANGES_REQUESTED** [review]
Some minor comments here and there, overall awesome work.

**Review by nikhen-s — APPROVED** [review]
Just approving so after changes, you can directly merge to main! Thank you!

File: template/9-design.md
Comment (achillet3) [reply template/9-design.md]: You're right! Everything was corrected

File: template/9-design.md
Comment (achillet3) [reply template/9-design.md]: I have taken it out as there shouldn't too much issues in the beginning indeed

File: template/9-design.md
Comment (achillet3) [reply template/9-design.md:28]: Added it as an extra Essential Screen and explained that it was at the bottom of all other screens

File: template/9-design.md
Comment (achillet3) [reply template/9-design.md:14]: You're right! Added a third-party payment SDK to integrate this

File: template/9-design.md
Comment (achillet3) [reply template/9-design.md:80]: Modified as well to take that into account

**Review by wisdomseeker2001 — APPROVED** [review]

**Pull request closed without merging**
